### PR TITLE
feat(userCtrl): implement db server sign up model and request handler

### PR DESCRIPTION
**Pull request by arminsebastian**

### CODE REVIEW

File: client/src/components/requester_task/partial-location.html [hunk 1]
```diff
@@ -4,15 +4,12 @@
...
   <div id="map">
-    this is a map
-    <div id="shop" ng-repeat="shop in ctrl.activeShops">
-      <div ng-click= "ctrl.createMission(shop)">
-        {{ ctrl.shop | json }}
-      </div>
+  </div>
+  <div id="shop" ng-repeat="shop in ctrl.activeShops">
+    <div ng-click= "ctrl.selectLocation(shop)">
+     Click Chossen Location: {{ shop | json }}
```
Comment (darrinmn9) [inline client/src/components/requester_task/partial-location.html]: change "Click Chossen Location" to "Select Location"

File: client/src/components/signup/signup.html [hunk 2]
```diff
@@ -13,17 +14,19 @@
     Expiration Date: <input type="text" ng-model="ctrl.user.expDate" /><br/>
     Billing Address: <input type="text" ng-model="ctrl.user.billingAddress" /><br/>
     City: <input type="text" ng-model="ctrl.user.city" /><br/>
-    <select name="URL" onchange="">
+    <select name="URL" ng-model="ctrl.user.state">
       <option selected="selected" value="">Select State</option>
-      <option value="">Random value</option>
+      <option  value="California">California</option>
```
Comment (tats-tagawa) [inline client/src/components/signup/signup.html:19]: extra space between `option` and `value`

also on 24 and 25

File: server/requesters/requesterCtrl.js [hunk 3]
```diff
@@ -15,23 +14,24 @@ module.exports = {
...
       status: status
     };
 
     //TODO: (db) insert a new transaction with order details into transactions
+
     TransactionCtrl.createTransaction(order);
 
     //FIX: change response to be id of newly created transaction
     console.log('Transaction was created!');
     res.sendStatus(201).send(1);
```
Comment (darrinmn9) [inline server/requesters/requesterCtrl.js]: is there a reason we are sending 1 ? or can we just do .send() ?

File: server/users/userCtrl.js [hunk 4]
```diff
@@ -11,5 +11,32 @@ module.exports = {
...
+      phoneNumber: reqUser.phoneNumber,
+      cardNumber: reqUser.creditCard,
+      cvc: reqUser.cvc,
+      expirationDate: reqUser.expDate,
+      billingAddress: reqUser.billingAddress,
+      city: reqUser.city,
+      state: reqUser.state,
+      country: reqUser.country
+    });
+    newUser.save(function(err){
```
Comment (tats-tagawa) [inline server/users/userCtrl.js:32]: no space before `{`
should be `function(err) {`

File: server/users/userCtrl.js [hunk 5]
```diff
@@ -11,5 +11,32 @@ module.exports = {
...
+      if(err){
+        console.log(err);
+      }
+    }).then(function() {
+      console.log('DB:', 'saved');
+      res.status(201).send('User created');
```
Comment (darrinmn9) [inline server/users/userCtrl.js:38]: no need to send anything back to client

File: server/users/userCtrl.js [hunk 6]
```diff
@@ -11,5 +11,32 @@ module.exports = {
...
+      cardNumber: reqUser.creditCard,
+      cvc: reqUser.cvc,
+      expirationDate: reqUser.expDate,
+      billingAddress: reqUser.billingAddress,
+      city: reqUser.city,
+      state: reqUser.state,
+      country: reqUser.country
+    });
+    newUser.save(function(err){
+      if(err){
```
Comment (tats-tagawa) [inline server/users/userCtrl.js:33]: missing space before `(` and after `)`
should be `if (err) {`

**Pull request merged**